### ISSUE TITLE
cppcheck preprocessed code with variadic can't be compiled

### ISSUE THREAD

**Issue by IOBYTE** [danmar/simplecpp#175]
Cppcheck -E converts code with `...` to `. . .` which then can't be compiled.

This code:
```
template<typename T, typename ... Args>
void log(T first, Args ... args) {
    log(args ...);
}
```
gets preprocessed to:
```
template < typename T , typename . . . Args >
void log ( T first , Args . . . args ) {
log ( args . . . ) ;
}
```
which produces this when compiled:
```
$ g++ logx.1.cpp
logx.1.cpp:1:34: error: expected nested-name-specifier before ‘.’ token
 template < typename T , typename . . . Args >
                                  ^
logx.1.cpp:1:34: error: expected ‘>’ before ‘.’ token
logx.1.cpp:2:22: error: ‘Args’ has not been declared
 void log ( T first , Args . . . args ) {
                      ^~~~
logx.1.cpp:2:27: error: expected ‘,’ or ‘...’ before ‘.’ token
 void log ( T first , Args . . . args ) {
                           ^
logx.1.cpp: In function ‘void log(T, int)’:
logx.1.cpp:3:7: error: ‘args’ was not declared in this scope
 log ( args . . . ) ;
       ^~~~
logx.1.cpp:3:14: error: expected unqualified-id before ‘.’ token
 log ( args . . . ) ;
              ^
logx.1.cpp:3:16: error: expected unqualified-id before ‘.’ token
 log ( args . . . ) ;
                ^
logx.1.cpp:3:18: error: expected unqualified-id before ‘)’ token
 log ( args . . . ) ;
                  ^
```

**Comment by IOBYTE** [comment]
I'm working on this. I have everything but the ast in cppcheck working.

The simplecpp changes are minor but the cppcheck changes are not.

**Comment by danmar** [comment]
nice

**Comment by IOBYTE** [comment]
Here is a PR that fixes this: https://github.com/danmar/simplecpp/pull/176

The simplecpp.cpp change is also in the cppcheck PR: https://github.com/danmar/cppcheck/pull/2143

I'm not sure how to keep the 2 in sync so I duplicated them.

**Comment by IOBYTE** [comment]
Fixed by: 28f2dee07281e2dcf1f31159eb6358da5ee2c7c1
